### PR TITLE
Add category feature

### LINKED ISSUE #27: [RFC] Ability to deal with "categories"
**Problem:** The current registry is quite simple but in complex applications, we use to have many dictionaries. The possibility to have a tree for splitting dictionaries in categories and subcategories would be a great improvement.

**Solution:** I suggest to add a new kind of registry and the notion of "category".

### PR DESCRIPTION
**Pull request by Nek-**
Fix #27

### CODE REVIEW

**Comment by athos7933** [comment]
Maybe need to update readme ?

**Comment by Nek-** [comment]
@athos7933 is right. It makes me feel it misses configuration as well.

**Comment by Nek-** [comment]
Looks ready to me :). Waiting for feedbacks.

File: README.md [hunk 1]
```diff
@@ -123,6 +123,28 @@ knp_dictionary:
 Callable dictionaries are loaded with a lazy strategy. It means that the callable
 will not be called if you do not use the dictionary.
 
+### Categories in dictionaries
+
+You can specity a category to your dictionary so you can get only some dictionaries later.
```
Comment (Awkan) [inline README.md]: `specify`

File: src/Knp/DictionaryBundle/Dictionary/Factory/CallableFactory.php [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 namespace Knp\DictionaryBundle\Dictionary\Factory;
 
+use Grpc\Call;
```
Comment (Valouleloup) [inline src/Knp/DictionaryBundle/Dictionary/Factory/CallableFactory.php]: What ?

**Pull request merged**